src/app/Events/EventTemplates.tsx

```diff
@@ -214,68 +223,66 @@ export const EventTemplates: React.FunctionComponent<EventTemplatesProps> = (pro
-        t.type.charAt(0).toUpperCase() + t.type.slice(1).toLowerCase(),
-      ]),
-    [filteredTemplates]
-  );
-
   const handleDelete = React.useCallback(
-    (rowData) => {
+    (t: EventTemplate) => {
       addSubscription(
         context.api
-          .deleteCustomEventTemplate(rowData[0])
+          .deleteCustomEventTemplate(t.name)
           .pipe(first())
           .subscribe(() => {} /* do nothing - notification will handle updating state */)
       );
     },
     [addSubscription, context.api]
   );
 
-  const actionResolver = (rowData: IRowData, extraData: IExtraData) => {
-    if (typeof extraData.rowIndex == 'undefined') {
-      return [];
-    }
-    let actions = [
-      {
-        title: 'Create Recording...',
-        onClick: (event, rowId, rowData) =>
-          history.push({
-            pathname: '/recordings/create',
-            state: { template: rowData[0], templateType: String(rowData[3]).toUpperCase() },
-          }),
-      },
-    ] as IAction[];
+  const handleDeleteButton = React.useCallback(
+    (t: EventTemplate) => {
+      if (context.settings.deletionDialogsEnabledFor(DeleteWarningType.DeleteEventTemplates)) {
+        setTemplateToDelete(t);
+        setWarningModalOpen(true);
+      } else {
+        handleDelete(t);
+      }
+    },
+    [context, context.settings, setWarningModalOpen, setTemplateToDelete, handleDelete]
+  );
 
-    const template: EventTemplate = filteredTemplates[extraData.rowIndex];
-    if (template.name !== 'ALL' || template.type !== 'TARGET') {
-      actions = actions.concat([
+  const actionsResolver = React.useCallback(
+    (t: EventTemplate) => {
+      let actions = [
         {
-          title: 'Download',
-          onClick: (event, rowId) => context.api.downloadTemplate(filteredTemplates[rowId]),
+          title: 'Create Recording...',
+          onClick: () =>
+            history.push({
+              pathname: '/recordings/create',
+              state: { templateName: t.name, templateType: t.type } as CreateRecordingProps,
+            }),
         },
-      ]);
-    }
-    if (template.type === 'CUSTOM') {
-      actions = actions.concat([
-        {
-          isSeparator: true,
-        },
-        {
-          title: 'Delete',
-          onClick: (event, rowId, rowData) => {
-            handleDeleteButton(rowData);
+      ] as IAction[];
+
+      if (t.name !== 'ALL' || t.type !== 'TARGET') {
+        actions = actions.concat([
+          {
+            title: 'Download',
+            onClick: () => context.api.downloadTemplate(t),
           },
-        },
-      ]);
-    }
-    return actions;
-  };
+        ]);
+      }
+      if (t.type === 'CUSTOM') {
+        actions = actions.concat([
+          {
+            isSeparator: true,
+          },
+          {
+            title: 'Delete',
+            onClick: () => handleDeleteButton(t),
+          },
+        ]);
+      }
+      return actions;
+    },
+    [context.api, history]
```

Comment:
oh yes definitely! I will add it in. Pretty easy to miss those deps :D

Reply:
I think there is or was something in the lint config that would pick these things up and automatically warn about them.

It's long past due that I (or someone) goes through and cleans up all the current linting problems so that the linting can be turned on for CI and added to the usual development workflow...